include/deal.II/grid/connectivity.h

```diff
@@ -0,0 +1,1477 @@
+          if (i == std::array<T, 3>{{j[0], j[2], j[1]}}) return 0;
+
+          // face_orientation=false, face_rotation=true, face_flip=false
+          if (i == std::array<T, 3>{{j[1], j[2], j[0]}}) return 2;
+
+          // face_orientation=false, face_rotation=false, face_flip=true
+          if (i == std::array<T, 3>{{j[2], j[1], j[0]}}) return 4;
+          // clang-format on
+        }
+      else if (type == 3) // QUAD
+        {
+          const std::array<T, 4> i{var_0[0], var_0[1], var_0[2], var_0[3]};
+          const std::array<T, 4> j{var_1[0], var_1[1], var_1[2], var_1[3]};
+
+          // clang-format off
+          // face_orientation=true, face_rotation=false, face_flip=false
+          if (i == std::array<T, 4>{{j[0], j[1], j[2], j[3]}}) return 1;
+
+          // face_orientation=true, face_rotation=true, face_flip=false
+          if (i == std::array<T, 4>{{j[1], j[3], j[0], j[2]}}) return 3;
+
+          // face_orientation=true, face_rotation=false, face_flip=true
+          if (i == std::array<T, 4>{{j[3], j[2], j[1], j[0]}}) return 5;
+
+          // face_orientation=true, face_rotation=true, face_flip=true
+          if (i == std::array<T, 4>{{j[2], j[0], j[3], j[1]}}) return 7;
+
+          // face_orientation=false, face_rotation=false, face_flip=false
+          if (i == std::array<T, 4>{{j[0], j[2], j[1], j[3]}}) return 0;
+
+          // face_orientation=false, face_rotation=true, face_flip=false
+          if (i == std::array<T, 4>{{j[2], j[3], j[0], j[1]}}) return 2;
+
+          // face_orientation=false, face_rotation=false, face_flip=true
+          if (i == std::array<T, 4>{{j[3], j[1], j[2], j[0]}}) return 4;
+
+          // face_orientation=false, face_rotation=true, face_flip=true
+          if (i == std::array<T, 4>{{j[1], j[0], j[3], j[2]}}) return 6;
+          // clang-format on
+        }
+
+      AssertThrow(false, dealii::StandardExceptions::ExcNotImplemented());
+
+      return -1;
+    }
+
+
+
+    /**
+     * Build entities of dimension d (with 0<d<dim). Entities are described by
+     * a set of vertices.
+     *
+     * Furthermore, the function determines for each cell of which d-dimensional
+     * entity it consists of and its orientation relative to the cell.
+     */
+    template <int key_length, typename FU>
+    void
+    build_entity_templated(
+      const unsigned int                                d,
+      const std::vector<std::shared_ptr<CellTypeBase>> &cell_types,
+      const std::vector<unsigned int> &                 cell_types_index,
+      const CRS<unsigned int> &                         crs,
+      CRS<unsigned int> &                               crs_d,        // result
+      CRS<unsigned int> &                               crs_0,        // result
+      std::vector<unsigned char> &                      orientations, // result
+      const FU &                                        second_key_function)
+    {
+      const bool comptibility_mode = true;
```

Comment:
You probably want to fix this typo (I was looking for the variable and couldn't find it in the usual way):
```suggestion
      const bool compatibility_mode = true;
```